src/loaders/compute_tools/microtrait/microtrait_util.py

```diff
@@ -39,6 +86,18 @@ def create_trait_unwrapped_rules(
             trait_name = row[trait_name_col]
             trait_rule_unwrapped_mapping[trait_name].add(rule_name) if pd.notna(trait_name) else None
 
+    trait_substrate_mapping = _retrieve_trait_substrate_mapping(substrate2rule_file)
+
+    # find the rule names associated with the substrate names from the rule2trait file
+    for trait_name, substrate_names in trait_substrate_mapping.items():
+        for substrate_name in substrate_names:
+            # filter the rule2trait DataFrame to include only rows where the substrate column contains the specified
+            # substrate name (the substrate column can contain multiple substrates i.e. "trehalose;maltose;sucrose")
+            filtered_df = rule2trait_df[
+                rule2trait_df[_RULE2TRAIT_RULE_SUBSTRATE_COL].str.contains(substrate_name, na=False)]
```

Comment:
Re the `na` argument, will there ever be a missing trait substrate?

Reply:
yes. in rule2trait file, some rows in substrate column is null.

Reply:
Can you add a comment to that effect

Reply:
👍

Reply:
:+1: